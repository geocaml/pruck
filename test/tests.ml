open Pruck
open Eio

module C = struct
  let area_name = Column.String "Area name"
  let freq = Column.String "Frequency"
  let year_2016 = Column.Float "2016"
  let year_2017 = Column.Float "2017"
  let year_2018 = Column.Float "2018"
  let year_2019 = Column.Float "2019"
  let year_2020 = Column.Float "2020"
  let year_2021 = Column.Float "2021"
  let year_2022 = Column.Float "2022"

  let all =
    Column.
      [
        area_name;
        freq;
        year_2016;
        year_2017;
        year_2018;
        year_2019;
        year_2020;
        year_2021;
        year_2022;
      ]
end

let main path =
  let df = Df.read_csv ~fill_default:true ~columns:C.all path in
  Df.where df C.area_name (String.equal "Cambridge");
  Fmt.pr "%a" Df.pp df

let () =
  Eio_main.run @@ fun env ->
  let cwd = Stdenv.cwd env in
  main Path.(cwd / "gov-uk-cycling.csv")
