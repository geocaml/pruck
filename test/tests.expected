<9 columns x 2 rows>
Area name : string, Frequency : string, 2016 : float, 2017 : float, 2018 : float, 2019 : float, 2020 : float, 2021 : float, 2022 : float
Cambridge, At least once per month, 63.2, 60.7, 63.7, 60.3, 51.2, 49.9, 55.9
Cambridge, At least once per week, 56.9, 54.2, 57.5, 55.2, 44.8, 42.6, 49.8